internal/ingress/election/election.go

```diff
@@ -76,11 +99,12 @@ func NewElector(config Config) Elector {
 	ttl := 30 * time.Second
 	le, err := leaderelection.NewLeaderElector(
 		leaderelection.LeaderElectionConfig{
-			Lock:          &lock,
-			LeaseDuration: ttl,
-			RenewDeadline: ttl / 2,
-			RetryPeriod:   ttl / 4,
-			Callbacks:     config.Callbacks,
+			Lock:            &lock,
+			LeaseDuration:   ttl,
+			RenewDeadline:   ttl / 2,
+			RetryPeriod:     ttl / 4,
+			Callbacks:       config.Callbacks,
+			ReleaseOnCancel: true,
```

Comment:
My testing today shows that this option does not work as intended: the _ConfigMap_ does not get updated when the `context.Contex` passed to `(*LeaderElector).Run` is done. There's no failure reported in the log. I'm still investigating what could be causing that.

Reply:
By way of using the Go modules `replace` directive and instrumenting the _client-go_ library, I figured out that this _does_ work—when the process doesn't die too soon. I spent about five hours today experimenting with changes to the shutdown sequence, and had to stop my work for the day before I could offer a proven solution. More detail tomorrow.

Reply:
It came down to how the `main` function was arranged. `main` blocked on `(*KongController).Start`. The signal handler started by `main` calls `(*KongController).Stop`. When `Stop` closes the the `KongController.stopCh` channel, `(*KongController).Start` returns. That allowed `main` to return before `(*KongController).Stop` had completed, interrupting the various intended termination procedures.

I revised how `main` interacts with the `handleSigterm` function. Rather than having the latter terminate the program, it instead indicates that it's complete by sending a recommended exit code to a channel. `main` blocks on receiving that exit code and exits when it's sure that it's safe to do so.